api/ads.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+import "api/base.proto";
+import "api/cds.proto";
+import "api/eds.proto";
+import "api/lds.proto";
+import "api/rds.proto";
+
+import "google/api/annotations.proto";
+
+// See https://github.com/lyft/envoy-api#apis for a description of the role of
+// ADS and how it is intended to be used by a management server. ADS requests
+// have the same structure as their singleton xDS counterparts, but can
+// multiplex many resource types on a single stream. The type_url in the
+// DiscoveryRequest/DiscoveryResponse provides sufficient information to recover
+// the multiplexed singleton APIs at the Envoy instance and management server.
+service AggregatedDiscoveryService {
+  // This is a gRPC-only API.
+  rpc StreamAggregatedResources(stream DiscoveryRequest)
+      returns (stream DiscoveryResponse) {
```

Comment:
I considered this and opted for simplicity in this PR, but I'm open to adopting the atomic list concept if we can provide a concrete use case. By moving to lists of updates in each response we also make the requests more complicated, as they need to express multiple version numbers (for each singleton xDS API that gets updated or fails its update). I think it's worth doing if we have a strong motivator here, but all things being equal I'm tending towards simplicity over future proofing.

There's a nice property of the existing API that a server that can handle ADS requests can also handle arbitrary singleton xDS requests as well, if we add more structure to the ADS request then this goes away, but maybe it's not that important.